### PR TITLE
Serve refinements

### PR DESCRIPTION
**Pull request by deanwampler**
## Why are these changes needed?

Provides a little more clarification on the requirements for state handling in the backend handlers for Ray Serve.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Review by simon-mo — APPROVED** [review]
LGTM

File: doc/source/rayserve/overview.rst [hunk 1]
```diff
@@ -120,36 +122,43 @@ Note that this will not delete any associated backends, which can be reused for
...
+Once you define the function (or class) that will handle a request.
 You'd use a function when your response is stateless and a class when you
-might need to maintain some state (like a model). 
-For both functions and classes (that take as input Flask Requests), you'll need to 
+might need to maintain some state (like a model).
+For both functions and classes (that take as input Flask Requests), you'll need to
 define them as backends to RayServe.
 
-It's important to note that RayServe places these backends in individual workers, which are replicas of the model.
+It's important to note that RayServe places these backends in individual workers, which are replicas of the model. This means that any global state you need shared across workers must be in a Ray actor (or external data store), which is referenced from the function or regular class you specify for the backend.
```
Comment (edoakes) [inline doc/source/rayserve/overview.rst:135]:
```suggestion
It's important to note that RayServe places these backends in individual worker processes, which are replicas of the model. This means that any global state you need shared across workers must be in a Ray actor (or external data store), which is referenced from the function or regular class you specify for the backend.
```

File: doc/source/rayserve/overview.rst [hunk 2]
```diff
@@ -120,36 +122,43 @@ Note that this will not delete any associated backends, which can be reused for
...
 
-  serve.create_backend("simple_backend", handle_request)
-  serve.create_backend("simple_backend_class", RequestHandler)
+  serve.create_backend("simple_backend", RequestHandler)
 
-Lastly, we need to link the particular backend to the server endpoint. 
+Lastly, we need to link the particular backend to the server endpoint.
 To do that we'll use the ``link`` capability.
-A link is essentially a load-balancer and allow you to define queuing policies 
+A link is essentially a load-balancer and allow you to define queuing policies
```
Comment (edoakes) [inline doc/source/rayserve/overview.rst:161]: `link` doesn't exist anymore. While you're at it, could you update these lines to say `set_traffic`?

File: doc/source/rayserve/overview.rst (hunk 1 above)
Comment (edoakes) [inline doc/source/rayserve/overview.rst:135]:
```suggestion
It's important to note that RayServe places these backends in individual workers, which are replicas of the model. If you need to share state between individual replicas, this must be done externally (e.g., through a Ray actor or external storage).
```

**Review by edoakes — COMMENTED** [review]
LGTM, just a few small comments

**Comment by simon-mo** [comment]
@deanwampler can you rebase and merge the changes? Thanks!

**Pull request closed without merging**